core/common/src/Utf8.kt

```diff
@@ -196,24 +205,29 @@ public fun Source.readString(byteCount: Long): String {
 }
 
 /**
- * Removes and returns a single UTF-8 code point, reading between 1 and 4 bytes as necessary.
+ * Decodes a single code point value from UTF-8 code units, reading between 1 and 4 bytes as necessary.
  *
  * If this source is exhausted before a complete code point can be read, this throws an
  * [EOFException] and consumes no input.
  *
- * If this source doesn't start with a properly-encoded UTF-8 code point, this method will remove
+ * If this source doesn't start with a properly encoded UTF-8 code point, this method will remove
  * 1 or more non-UTF-8 bytes and return the replacement character (`U+fffd`). This covers encoding
```

Comment:
Given that we write `?` instead of a proper replacement char on encoding, shouldn't we do the same for decoding?

Reply:
Currently, such a behavior is consistent with how `readString`/`writeString` works.

Given that the problem also arose around `utf8Size`, I think it should be resolved separately to take various options and opinions into account.